### PR TITLE
Moving from circleci to gha WRT benchmark flows

### PR DESCRIPTION
**Pull request by filipecosta90**
This PR removes the usage of circleci for benchmark workflows. It introduces 3 github actions:
- One for merges to version and master/main branches
- One for triggering benchmarks during PR work
- One for triggering nightly benchmarks twice a week

It requires the following secrets to be present among the repo action secrets:
- PERFORMANCE_RTS_AUTH
- PERFORMANCE_RTS_HOST
- PERFORMANCE_RTS_PORT
- PERFORMANCE_WH_TOKEN
- PERFORMANCE_GH_TOKEN
- PERFORMANCE_EC2_ACCESS_KEY
- PERFORMANCE_EC2_SECRET_KEY
- PERFORMANCE_EC2_REGION
- PERFORMANCE_EC2_PRIVATE_PEM

Checks:
- [x] confirm secrets are present
- [x] confirm comment with performance info is present on this PR: https://github.com/RedisBloom/RedisBloom/pull/683#issuecomment-1752404644

### CODE REVIEW

**Comment by filipecosta90** [comment]
### Automated performance analysis summary

This comment was automatically generated given there is performance data available.

In summary:
- Detected a total of 16 stable tests between versions.
- Detected a total of 1 improvements above the improvement water line.
- Detected a total of 2 regressions bellow the regression water line 5.0.

You can check a comparison in detail via the [grafana link](https://benchmarksrediscom.grafana.net/d//q4-5sRR7k/?var-branch=master&var-branch=benchmarks.gha)

### Comparison between master and benchmarks.gha.

Time Period from 30 days ago. (environment used: oss-standalone)

|                     Test Case                      |Baseline master (median obs. +- std.dev)|Comparison benchmarks.gha (median obs. +- std.dev)|% change (higher-better)|                Note                 |
|----------------------------------------------------|----------------------------------------|--------------------------------------------------|------------------------|-------------------------------------|
|bf_add_cap10M_err0.001                              | 144921 +- 4.6%  (5 datapoints)         | 157473 +- nan%  (1 datapoints)                   |8.7%                    |IMPROVEMENT                          |
|cf_add_cap10M                                       | 156243 +- 6.4%  (5 datapoints)         | 148692 +- nan%  (1 datapoints)                   |-4.8%                   |waterline=6.4%. potential REGRESSION |
|cms_add_err0.001_prob0.01                           | 154433 +- 5.7%  (5 datapoints)         | 144398 +- nan%  (1 datapoints)                   |-6.5%                   |waterline=5.7%. REGRESSION           |
|tdigest_add_comp100                                 | 152665 +- 7.7%  (5 datapoints)         | 158723 +- nan%  (1 datapoints)                   |4.0%                    |waterline=7.7%. potential IMPROVEMENT|
|tdigest_add_comp1000                                | 145979 +- 7.6%  (5 datapoints)         | 149245 +- nan%  (1 datapoints)                   |2.2%                    |waterline=7.6%. -- no change --      |
|tdigest_add_comp200                                 | 150369 +- 5.7%  (5 datapoints)         | 145448 +- nan%  (1 datapoints)                   |-3.3%                   |waterline=5.7%. potential REGRESSION |
|tdigest_add_comp50                                  | 154433 +- 7.0%  (5 datapoints)         | 155635 +- nan%  (1 datapoints)                   |0.8%                    |waterline=7.0%. -- no change --      |
|tdigest_add_comp500                                 | 148139 +- 6.1%  (5 datapoints)         | 138881 +- nan%  (1 datapoints)                   |-6.2%                   |waterline=6.1%. REGRESSION           |
|tdigest_merge_2hist_1M_samples_comp100              | 70002 +- 4.2%  (5 datapoints)          | 71551 +- nan%  (1 datapoints)                    |2.2%                    |-- no change --                      |
|tdigest_quantile_p50_1M_samples_comp100             | 130534 +- 3.0%  (5 datapoints)         | 132028 +- nan%  (1 datapoints)                   |1.1%                    |-- no change --                      |
|tdigest_quantile_p50_1M_samples_comp1000            | 134371 +- 2.4%  (5 datapoints)         | 134815 +- nan%  (1 datapoints)                   |0.3%                    |-- no change --                      |
|tdigest_quantile_p50_1M_samples_comp200             | 133599 +- 6.1%  (5 datapoints)         | 136249 +- nan%  (1 datapoints)                   |2.0%                    |waterline=6.1%. -- no change --      |
|tdigest_quantile_p50_1M_samples_comp50              | 133536 +- 3.9%  (5 datapoints)         | 135229 +- nan%  (1 datapoints)                   |1.3%                    |-- no change --                      |
|tdigest_quantile_p50_1M_samples_comp500             | 132957 +- 3.2%  (5 datapoints)         | 134898 +- nan%  (1 datapoints)                   |1.5%                    |-- no change --                      |
|tdigest_quantile_p50_p95_p99_p999_1M_samples_comp100| 121818 +- 5.0%  (5 datapoints)         | 120751 +- nan%  (1 datapoints)                   |-0.9%                   |waterline=5.0%. -- no change --      |
|tdigest_quantile_p95_1M_samples_comp100             | 135010 +- 2.8%  (5 datapoints)         | 134736 +- nan%  (1 datapoints)                   |-0.2%                   |-- no change --                      |
|tdigest_quantile_p99_1M_samples_comp100             | 132891 +- 2.3%  (5 datapoints)         | 137358 +- nan%  (1 datapoints)                   |3.4%                    |potential IMPROVEMENT                |
|tdigest_trimmed_mean_1M_samples_comp100             | 134801 +- 3.2%  (5 datapoints)         | 135136 +- nan%  (1 datapoints)                   |0.2%                    |-- no change --                      |
|topk_incrby_k100                                    | 147052 +- 5.2%  (5 datapoints)         | 148142 +- nan%  (1 datapoints)                   |0.7%                    |waterline=5.2%. -- no change --      |

File: .github/workflows/trigger-benchmark.yml [hunk 1]
```diff
@@ -18,17 +18,7 @@ jobs:
         with:
           contains: 'run-benchmark'
   perf-ci:
-    name: Trigger CI benchmarks
+    name: Trigger CI benchmarks from label
     needs: haslabel
-    if: needs.haslabel.outputs.benchmark
```
Comment (GuyAv46) [inline .github/workflows/trigger-benchmark.yml:23]: Removed by mistake?

File: .github/workflows/nightly-benchmark.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: Run nightly CI benchmarks
+
+on:
+  push:
+    branches:
+      - master
+      - 2.6
```
Comment (GuyAv46) [inline .github/workflows/nightly-benchmark.yml:7]: Don’t we want a regex for version branches?

**Review by GuyAv46 — DISMISSED** [review]

File: .github/workflows/nightly-benchmark.yml (hunk 2 above)
Comment (filipecosta90) [reply .github/workflows/nightly-benchmark.yml:7]: indeed. addressed in the latest comment

File: .github/workflows/trigger-benchmark.yml (hunk 1 above)
Comment (filipecosta90) [reply .github/workflows/trigger-benchmark.yml:23]: yes. addressed in latest comment

**Review by GuyAv46 — APPROVED** [review]

**Pull request merged**